### DISCUSSION TITLE
How to set certs (mqttS) through env vars?

### DISCUSSION THREAD

**Discussion by klemen-df** [emqx/emqx#10335] [Q&A] [answered]
I've succesfully started mqtts through env vars:

`EMQX_LISTENERS__SSL__MQTTS__BIND=0.0.0.0:8883
`

```
kubectl exec -it emqx-core-0 -c emqx -- emqx_ctl listeners
ssl:mqtts
  listen_on       : 0.0.0.0:8883
  acceptors       : 16
  proxy_protocol  : false
  running         : true
  current_conn    : 3
  max_conns       : 1024000
```

**How do I now configure certs and tls versions through environment variables?**

As written here and [here](https://www.emqx.io/docs/en/v5/security/ssl.html#ssl-listener-for-mqtt)  it should be something like:

- 'listeners.ssl.mqtts.ssl_options.keyfile' ( 
- 'listeners.ssl.mqtts.ssl_options.certfile' ( 
- 'listeners.ssl.mqtts.ssl_options.cacertfile'
- 'listeners.ssl.mqtts.ssl_options.versions'

According to HOCON this makes it 'EMQX_LISTENERS__SSL__MQTTS__SSL_OPTIONS_CACERTFILE' but it doesn't work - unknown variable.

**Comment by id** [comment, marked as answer]
I think it should be `EMQX_LISTENERS__SSL__MQTTS__SSL_OPTIONS__CACERTFILE`, i.e. `.` --> `__`

**Comment by klemen-df** [comment]
Oh yes, thanks!
